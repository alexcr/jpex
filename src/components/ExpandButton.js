import React from 'react';
import styled from 'styled-components';

const Button = styled.div`
  width: 20px;
  margin-right: 10px;
  display: inline-block;
  text-align: center;
  color: #01b227;
  cursor: pointer;

  &:hover {
    color: #006616;
  }
`;

export function ExpandButton({
  isExpanded,
  onToggle,
  isArray,
}) {
  return (
    <Button onClick={onToggle}>
      {
        isArray ?
        (isExpanded ? '[-]' : '[+]') :
        (isExpanded ? '{-}' : '{+}')
      }
    </Button>
  );
}
