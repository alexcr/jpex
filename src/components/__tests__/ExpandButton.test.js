import React from 'react';
import { shallow } from 'enzyme';
import { ExpandButton } from '../ExpandButton';

const clickFn = jest.fn();

describe('ExpandButton', () => {
  it('should render the expand array button correctly', () => {
    expect(
      shallow(<ExpandButton isExpanded={false} isArray={true} />)
    ).toMatchSnapshot();
  });

  it('should render the expand object button correctly', () => {
    expect(
      shallow(<ExpandButton isExpanded={false} isArray={false} />)
    ).toMatchSnapshot();
  });

  it('should render the collapse array button correctly', () => {
    expect(
      shallow(<ExpandButton isExpanded={true} isArray={true} />)
    ).toMatchSnapshot();
  });

  it('should render the collapse object button correctly', () => {
    expect(
      shallow(<ExpandButton isExpanded={true} isArray={false} />)
    ).toMatchSnapshot();
  });

  it('should call the onToggle callback when clicked', () => {
    shallow(<ExpandButton onToggle={clickFn} />).simulate('click');
    expect(clickFn).toHaveBeenCalled();
  });
});
