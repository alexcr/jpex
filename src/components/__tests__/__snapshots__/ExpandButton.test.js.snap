// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExpandButton should render the collapse array button correctly 1`] = `
<styled.div>
  [-]
</styled.div>
`;

exports[`ExpandButton should render the collapse object button correctly 1`] = `
<styled.div>
  {-}
</styled.div>
`;

exports[`ExpandButton should render the expand array button correctly 1`] = `
<styled.div>
  [+]
</styled.div>
`;

exports[`ExpandButton should render the expand object button correctly 1`] = `
<styled.div>
  {+}
</styled.div>
`;
